sr_gui_change_controllers/src/sr_gui_change_controllers/change_controllers.py

```diff
@@ -51,64 +58,179 @@ def __init__(self, context):
+        try:
+            resp1 = list_controllers()
+        except rospy.ServiceException:
+            success = False
+
+        if success:
+            running_controllers = [c for c in resp1.controller if c.state == "running"]
+        else:
+            rospy.loginfo(
+                "Couldn't get list of controllers from controller_manager/list_controllers service")
+            return
+
+        running_traj_controllers = []
+        running_pos_controllers = []
+        running_teach_controllers = []
+        robot_names = ["rh_", "lh_", "ra_", "la_"]
+        current_robot_control = {
+            'rh_': None,
+            'lh_': None,
+            'ra_': None,
+            'la_': None
+        }
+        for robot_name in robot_names:
+            for controller in running_controllers:
+                if robot_name in controller.name:
+                    if "position_controller" in controller.name:
+                        current_robot_control[robot_name] = 1
+                        break
+                    elif "trajectory_controller" in controller.name:
+                        current_robot_control[robot_name] = 0
+                        break
+                    elif "effort_controller" in controller.name:
+                        current_robot_control[robot_name] = 2
+                        break
+
+        for robot_name, control in current_robot_control.items():
+            if control is not None:
+                self.update_current_controller_field(control, robot_name)
+
+    def update_current_controller_field(self, ctrl_type, robot_name):
+        if "rh_" == robot_name:
+            for button in range(len(self._rh_teach_buttons)):
+                if button == ctrl_type:
+                    self._rh_teach_buttons[button].setIcon(self.CONTROLLER_ON_ICON)
+                else:
+                    self._rh_teach_buttons[button].setIcon(self.CONTROLLER_OFF_ICON)
+        elif "lh_" == robot_name:
+            for button in range(len(self._lh_teach_buttons)):
+                if button == ctrl_type:
+                    self._lh_teach_buttons[button].setIcon(self.CONTROLLER_ON_ICON)
+                else:
+                    self._lh_teach_buttons[button].setIcon(self.CONTROLLER_OFF_ICON)
+        elif "ra_" == robot_name and ctrl_type != 2:
+            for button in range(len(self._ra_teach_buttons)):
+                if button == ctrl_type:
+                    self._ra_teach_buttons[button].setIcon(self.CONTROLLER_ON_ICON)
+                else:
+                    self._ra_teach_buttons[button].setIcon(self.CONTROLLER_OFF_ICON)
+        elif "la_" == robot_name and ctrl_type != 2:
+            for button in range(len(self._la_teach_buttons)):
+                if button == ctrl_type:
+                    self._la_teach_buttons[button].setIcon(self.CONTROLLER_ON_ICON)
+                else:
+                    self._la_teach_buttons[button].setIcon(self.CONTROLLER_OFF_ICON)
```

Comment:
should handle unknown robot name for robustness (e.g. logerr ubnknwon robot name)